mods/ra/maps/negotiations/negotiations.lua

```diff
@@ -0,0 +1,1237 @@
+--[[
+   Copyright (c) The OpenRA Developers and Contributors
+   This file is part of OpenRA, which is free software. It is made
+   available to you under the terms of the GNU General Public License
+   as published by the Free Software Foundation, either version 3 of
+   the License, or (at your option) any later version. For more
+   information, see COPYING.
+]]
+BadGuy = Player.GetPlayer("BadGuy")
+England = Player.GetPlayer("England")
+Greece = Player.GetPlayer("Greece")
+GoodGuy = Player.GetPlayer("GoodGuy")
+Neutral = Player.GetPlayer("Neutral")
+USSR = Player.GetPlayer("USSR")
+
+ForestTeam = { ForestPatroller1, ForestPatroller2, ForestPatroller3, ForestPatroller4, ForestPatroller5, ForestPatroller6, ForestPatroller7, ForestPatroller8 }
+BlockerTeam = { BlockerGuard1, BlockerGuard2, BlockerTank }
+SoutheastTurretTeam = { SoutheastTurretRifle1, SoutheastTurretRifle2, SoutheastTurretRifle3 }
+Hostages = { Hostage1, Hostage2, Hostage3, Hostage4, Hostage5 }
+
+TimerColor = HSLColor.White
+TimeRemaining = 0
+
+TanyaType = "e7.noautotarget"
+if Difficulty == "easy" then
+	TanyaType = "e7"
+end
+
+WorldLoaded = function()
+	Camera.Position = TanyaDrop.CenterPosition
+	SpawnPlayerCamera(TanyaDrop.Location, DateTime.Seconds(20))
+
+	Trigger.AfterDelay(1, function()
+		Media.PlaySpeechNotification(Greece, "TimerStarted")
+	end)
+
+	StartSoldier.Move(WestRoadRally.Location)
+	if Difficulty == "hard" then
+		StartSoldier.CallFunc(ReinforceStartSoldiers)
+	end
+
+	VillageBridge = GetBridge()
+	ReinforceHardTeams()
+	PrepareBadGuy()
+	ReinforceTanya()
+	CreateInitialTriggers()
+	CreateInitialObjectives()
+	LinkCooperativeObjectives()
+	CheckSovietDestruction()
+end
+
+CreateInitialObjectives = function()
+	InitObjectives(Greece)
+	if Difficulty == "hard" then
+		SaveAllHostages = AddPrimaryObjective(Greece, "keep-all-hostages-alive")
+	end
+	FreeTheHostages = AddPrimaryObjective(Greece, "free-hostages")
+	SaveAllHostages = SaveAllHostages or AddSecondaryObjective(Greece, "keep-all-hostages-alive")
+	SurviveUntilEnd = AddPrimaryObjective(England, "")
+	StopAllies = AddPrimaryObjective(USSR, "")
+end
+
+LinkCooperativeObjectives = function()
+	local greeceActor = Greece.GetActorsByType("player")[1]
+	Trigger.Clear(greeceActor, "OnPlayerWon")
```

Comment:
It has to do with the way objectives are set up to address https://github.com/OpenRA/OpenRA/pull/20975#pullrequestreview-1608535624 's bit about objective message spam. The mission has points where this can happen:

1. Greece completes a primary objective.
2. Greece has completed all primary objectives, and is considered a winner. This triggers the OnPlayerWon added by InitObjectives from `campaign.lua`.
3. Mission completion isn't triggered because this is a co-op mission under the hood, and their ally England needs to also win.
4. After a delay, Greece receives a new primary objective and the mission carries on.

At step 2, there would normally be the speech announcement "Mission accomplished." The options I saw were to remove the trigger from Greece, or put a rewritten InitObjectives into the mission. I figured the first option was less awkward. A little after this line, England's win hands Greece's the normal announcement.

Ideally, I'd just set primary objectives at the start and delay showing/announcing them in a more straightforward way, but I realize that's a bit niche.